### DISCUSSION TITLE
can't load json file in userdata to use jinja template

### DISCUSSION THREAD

**Discussion by dvbfonsbiemans** [cloudreach/sceptre#935] [Q&A]
i'm trying to use a json document to use as paramters for a jinja template. 
we use the following parameters:

```
sceptre_user_data:
  test: !file_contents test.json
```

and try to load this json: 
```
{
    "permissionsets": [
      {
        "admin": [
          {
            "groupid1": [
              "3333333",
              "44444444"
            ]
          }
        ]
      },
      {
        "readonly": [
          {
            "groupid2": [
              "1111111",
              "2222222"
            ]
          }
        ]
      }
    ]
  }
```

and then use it in the following part in a j2 template:
  ```
{% set test = sceptre_user_data.test%}
    {% for t in test %}
      {{ t }}

  {% endfor %}
```

the prolem we have with this is we get he userdata only as a String, we have not found a way to use it as a dict. 

can some one help?

**Comment by craighurley** [comment]
On a mobile, so just a quick reply. 

Have you tried using one of jinjas built in filters, e.g. tojson?

Pseudo code:

set test = seceptre... | tojson
loop over test.path

**Comment by dvbfonsbiemans** [comment]
yeah, i tried that, but it than says that the object doesn't have an attribute. `{% for t in test.permissionset %}
"'markupsafe.Markup object' has no attribute 'permissionset'"`

i tried several variaties, but my jinja is also not the best.

**Comment by zaro0508** [comment]
I believe file_contents resolver passes it's content as a string.  however if you pass in the json without using file_contents resolver then scepter should pass in the contents of sceptre_user_data as an object.  try this..

```
sceptre_user_data:
  test: {"permissionsets":[{"admin":[{"groupid1":["3333333","44444444"]}]},{"readonly":[{"groupid2":["1111111","2222222"]}]}]}
``` 

if you still prefer keeping the json in a separate file then you will probably need to make a code change to the file_contents resolver.

**Comment by zaro0508** [comment]
I have created a new `!file` resolver[1] to address this issue.  It adds the ability to pass json/yaml objects to sceptre parameters. Would you mind taking a look at it @ngfgrant and let me know whether you would like to accept it into the GH Sceptre org?

[1] https://github.com/Sage-Bionetworks-IT/sceptre-file-resolver

**Comment by ngfgrant** [comment]
Hey folks,

Thanks for progressing this - I think it would make sense to have this as a sceptre resolver, if you are willing to share. We could have this and `file_contents` run side by side with the intention of deprecating `file_contents`. 

Sound good?

**Comment by zaro0508** [comment]
The resolver has been moved to the sceptre org, https://github.com/Sceptre/sceptre-file-resolver
